src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useContext, useMemo, useRef } from 'react';
+import { Header } from '../Header/Header';
+import { UserWarning } from '../../UserWarning';
+import { USER_ID } from '../../api/todos';
+import { DispatchContext, StateContext } from '../../Store';
+import { TodoList } from '../TodoList/TodoList';
+import { Footer } from '../Footer/Footer';
+import classNames from 'classnames';
+
+export const TodoApp: React.FC = () => {
+  const { todos, errorMessage } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const timer = useRef(0);
+
+  timer.current = window.setTimeout(
+    () => dispatch({ type: 'setError', message: '' }),
+    3000,
+  );
+
+  const unCompletedTodos = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const handleOnClickDelete = () => {
+    clearTimeout(timer.current);
+    dispatch({ type: 'setError', message: '' });
+  };
+
+  if (!USER_ID) {
+    return <UserWarning />;
+  }
```

Comment:
```suggestion
  if (!USER_ID) return <UserWarning />
```

Reply:
prettier does not allow these changes to be made